.github/workflows/ubuntu_workflow.yml

```
@@ -41,17 +41,18 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-gradle-
 
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+      
       - id: build
         name: Gradle clean build
-        uses: gradle/gradle-build-action@v3.5.0
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           HEAD_REF: ${{ github.head_ref }}
-        with:
-          arguments: "clean build"
-
+        run: ./gradlew clean build
+      
   publish-scan-url:
-    needs: [ build ]
+    needs: [build]
```

Personally, I prefer: 
```suggestion
    needs:
     - build
```
It makes it easier to add additional items and preserve blame.